logprep/connector/http/input.py

```diff
@@ -0,0 +1,187 @@
+        self._messages.put({"message": data.decode("utf8")})
+
+
+class Server(uvicorn.Server):
+    """the uvicorn server"""
+
+    def install_signal_handlers(self):
+        pass
+
+    @contextlib.contextmanager
+    def run_in_thread(self):
+        """Context manager to run the server in a separate thread"""
+        thread = threading.Thread(target=self.run)
+        thread.start()
+        try:
+            while not self.started:
+                pass
+            yield
+        finally:
+            self.should_exit = True
+            thread.join()
+
+
+class HttpConnector(Input):
+    """Connector to accept log messages as http post requests"""
+
+    _messages: queue.Queue = queue.Queue()
+
+    _endpoint_registry: Mapping[str, HttpEndpoint] = {
+        "json": JSONHttpEndpoint,
+        "plaintext": PlaintextHttpEndpoint,
+        "jsonl": JSONLHttpEndpoint,
+    }
+
+    @define(kw_only=True)
+    class Config(Input.Config):
+        """Config for HTTPInput"""
+
+        uvicorn_config: Mapping[str, Union[str, int]] = field(
+            validator=[
+                validators.instance_of(dict),
+                validators.deep_mapping(
+                    key_validator=validators.in_(UVICORN_CONFIG_KEYS),
+                    value_validator=lambda x, y, z: True,
+                ),
+            ]
+        )
+        """Configure uvicorn server. For possible settings see `uvicorn settings page`_
+        
+        .. _uvicorn settings page: https://www.uvicorn.org/settings
+        """
+        endpoints: Mapping[str, str] = field(
+            validator=[
+                validators.instance_of(dict),
+                validators.deep_mapping(
+                    key_validator=validators.matches_re(r"^\/.+"),
+                    value_validator=validators.in_(["json", "plaintext", "jsonl"]),
+                ),
+            ]
+        )
+        """Configure endpoint routes with a Mapping of an path to an endpoint. Possible endpoints
+        are: :code:`json`, :code:`jsonl`, :code:`plaintext`
+        """
+
+    app: FastAPI
+    server: uvicorn.Server
+
+    __slots__ = ["app", "server"]
+
+    def setup(self):
+        super().setup()
+        self.app = FastAPI()
+        for endpoint_path, endpoint_name in self._config.endpoints.items():
+            endpoint_class = self._endpoint_registry.get(endpoint_name)
+            endpoint = endpoint_class(self._messages)
+            self.app.add_api_route(
+                path=f"{endpoint_path}", endpoint=endpoint.endpoint, methods=["POST"]
+            )
+        config = uvicorn.Config(
+            **self._config.uvicorn_config, app=self.app, log_level=self._logger.level
+        )
+        self.server = Server(config)
+
+    def _get_event(self, timeout: float) -> Tuple:
+        """returns the first message from the queue"""
+        try:
+            return self._messages.get(timeout=timeout), None
+        except queue.Empty:
+            return None, None
```

Comment:
I think we'll have a problem with the fact that the second return value is always none once someone wants to calculate an HMAC on the RAW-Message, as this input connector doesn't have a raw message. Could you double check the hmac tests please. At the moment they mock the get_event method such that we don't really test for the real return values. Also, once you are at it, could you also rename the `kafka_config` variable name in the hmac tests in the `base.py` file. Due to the generic nature of these base tests it can include all kinds of connector configs, not only kafka.

Reply:
renamed the kafka things to connector things in the tests

Reply:
I changed it and added a test for it. plz look for it